logictest: increase rate limits for tenants

asubiotto opened the pull request:
The default rate limits were causing extreme test slowness in some logic tests
(e.g. cascade).

Release note: None

Fixes #53040

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53142)
<!-- Reviewable:end -->

Commit:
logictest: increase rate limits for tenants

The default rate limits were causing extreme test slowness in some logic tests
(e.g. cascade).

Release note: None

ajwerner reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53142)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @asubiotto)

___
*[pkg/sql/logictest/logic.go, line 1335 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53142#-MFC2-7s6WHAGd7jL7ps:-MFC2-7s6WHAGd7jL7pt:b-5l67h3) ([raw file](https://github.com/cockroachdb/cockroach/blob/05e09013892168bca6545e2cdef945a033f98564/pkg/sql/logictest/logic.go#L1335)):*
> ```Go
> 
> 		// Open a connection to this tenant to set any cluster settings specified
> 		// by the test config.
> ```

I’m pretty certain you need to do this as the system tenant. It will then apply to all tenants. If you do it as the tenant I think it will either fail or noop 


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53142)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[pkg/sql/logictest/logic.go, line 1335 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53142#-MFC2-7s6WHAGd7jL7ps:-MFF5Uip7pg51chmX2b5:bxn0psx) ([raw file](https://github.com/cockroachdb/cockroach/blob/05e09013892168bca6545e2cdef945a033f98564/pkg/sql/logictest/logic.go#L1335)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I’m pretty certain you need to do this as the system tenant. It will then apply to all tenants. If you do it as the tenant I think it will either fail or noop 
</blockquote></details>

This connection is added to `connsForClusterSettingChanges` so the cluster settings are set as both the system tenant and tenant 10 since cluster settings semantics differ. By the way, you can set in-memory cluster settings on the tenant using a testing knob, which is useful for the distsql setting for example.

The rate limits below are set as the system tenant.


<!-- Sent from Reviewable.io -->

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53142#-:-MFFrEmB2i7U1OkT_13d:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
